analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/CollectionTracker.cs

```diff
@@ -65,6 +82,11 @@ public static CollectionConstraint ObjectCreationConstraint(ProgramState state,
             ? CollectionConstraint.Empty
             : CollectionConstraint.NotEmpty;
 
+    public static CollectionConstraint MethodReferenceConstraint(IMethodReferenceOperationWrapper operation) =>
+        operation.Instance is not null && AddMethods.Contains(operation.Method.Name)
+            ? CollectionConstraint.NotEmpty
+            : null;
```

Comment:
Here we don't even check if the `instance` is typed as one of the tracked collection types.
Is this intended? 
e.g. `randomType.Add`

Reply:
I'll keep a note for later, thanks!